src/ImageSharp.Drawing/Brushes/SolidBrush{TColor}.cs

```diff
@@ -72,13 +76,43 @@ public SolidBrushApplicator(TColor color)
             /// <returns>
             /// The color
             /// </returns>
-            public override TColor this[int x, int y] => this.color;
+            internal override TColor this[int x, int y] => this.color;
 
             /// <inheritdoc />
             public override void Dispose()
             {
                 // noop
             }
+
+            /// <inheritdoc />
+            internal override void Apply(float[] scanlineBuffer, int scanlineWidth, int offset, int x, int y)
+            {
+                DebugGuard.MustBeGreaterThanOrEqualTo(scanlineBuffer.Length, offset + scanlineWidth, nameof(scanlineWidth));
+
+                using (PinnedBuffer<float> buffer = new PinnedBuffer<float>(scanlineBuffer))
+                {
+                    BufferPointer<float> slice = buffer.Slice(offset);
+
+                    for (int xPos = 0; xPos < scanlineWidth; xPos++)
+                    {
+                        int targetX = xPos + x;
+                        int targetY = y;
+
+                        float opacity = slice[xPos];
+                        if (opacity > Constants.Epsilon)
```

Comment:
These ifs are distracting CPU branch predictors. Isn't it possible solve this with pure multiplications instead? Same for `PremultipliedLerp()` implementation.